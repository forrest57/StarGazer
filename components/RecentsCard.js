import { View, Text, StyleSheet } from 'react-native'
import tw from 'twrnc'

export default CardComponent = ({ text }) => (
  <View
    style={tw`bg-gray-700 p-4 my-2 rounded-md flex w-full  justify-center flex-row`}>
    <Text style={styles.buttonText}>{text}</Text>
  </View>
)

const styles = StyleSheet.create({
  buttonText: {
    color: 'gray',
    fontSize: 28,
    fontWeight: 'bold',
  },
})
