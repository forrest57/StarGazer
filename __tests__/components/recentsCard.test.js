import { render, screen } from '@testing-library/react-native'

import React from 'react'
import RecentsCard from '../../components/RecentsCard'
import renderer from 'react-test-renderer'

describe('<RecentsCard />', () => {
  it('renders at all', () => {
    const tree = renderer.create(<RecentsCard text={'lorem'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders passed props', () => {
    render(<RecentsCard text={'lorem'} />)
    expect(screen.getByText('lorem')).toBeTruthy()
  })
})
