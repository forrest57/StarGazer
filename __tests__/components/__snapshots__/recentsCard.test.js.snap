// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentsCard /> renders at all 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#374151",
      "borderRadius": 6,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    lorem
  </Text>
</View>
`;
